routers/repo/commit.go

```diff
@@ -107,14 +107,42 @@ func SearchCommits(ctx *context.Context) {
 	ctx.Data["PageIsCommits"] = true
 	ctx.Data["PageIsViewCode"] = true
 
-	keyword := strings.Trim(ctx.Query("q"), " ")
-	if len(keyword) == 0 {
+	query := strings.Trim(ctx.Query("q"), " ")
+	if len(query) == 0 {
 		ctx.Redirect(ctx.Repo.RepoLink + "/commits/" + ctx.Repo.BranchNameSubURL())
 		return
 	}
+
+	var keywords, authors, committers []string
```

Comment:
Could you create a new function named `NewSearchCommitsOptions(searchstring string) SearchCommitsOptions` to parse the content?

Reply:
Done